src/main/java/frc/team2412/robot/commands/shooter/ShootingWhileMovingV2Command.java

```diff
@@ -0,0 +1,55 @@
+package frc.team2412.robot.commands.shooter;
+
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.team2412.robot.subsystem.ShooterSubsystem;
+import static frc.team2412.robot.subsystem.ShooterSubsystem.ShooterConstants.*;
+import frc.team2412.robot.subsystem.TargetLocalizer;
+import frc.team2412.robot.util.ShooterDataDistancePoint;
+
+public class ShootingWhileMovingV2Command extends CommandBase {
+    private final ShooterSubsystem shooter;
+    private final TargetLocalizer localizer;
+
+    public ShootingWhileMovingV2Command(ShooterSubsystem shooter, TargetLocalizer localizer) {
+        this.shooter = shooter;
+        this.localizer = localizer;
+        addRequirements(shooter);
+    }
+
+    @Override
+    public void execute() {
+        if (!localizer.hasTarget()) {
+            return;
+        }
+
+        // get target location
+        // get current shot time
+        // Adjust target location by that shot time
+        // point at that and shoot at that new distance
+
+        Translation2d targetPosition = localizer.getTargetPosition();
+
+        double actualDistance = targetPosition.getNorm();
```

Comment:
Since we only use `targetPosition` for the length, can't we replace this with `localizer.getDistance()`?